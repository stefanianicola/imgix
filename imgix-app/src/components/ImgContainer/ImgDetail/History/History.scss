.wrap-history {
    position: relative;
    margin: 50px 0;
    .row-prev {
        .btn {
            color: #fff;
            background-color:#f97641;
            border: #f97641;
            font-weight: 100;
            padding: 0 10px;
            margin-left: 50px;
            i {
                padding-right: 10px;
            }
            &:focus {
                box-shadow: none;
            }
            &:hover{
                background-color: #D25135;
            }
        }
    }
    .download {
        color: #fff;
        background-color:#f97641;
        border: #f97641;
        font-weight: 100;
        padding: 0 10px;
        position: absolute;
        bottom: 2px;
        right: 50px;
        &:hover{
            background-color: #D25135;
        }
    }
}
