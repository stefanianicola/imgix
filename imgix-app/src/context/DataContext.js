import React, { createContext, useState, useEffect } from 'react';
import axios from 'axios';
import Params from '../Data/Params';
export const DataContext = createContext();


const DataContextProvider = (props) => {
    const [data, setData] = useState([]);
    const [flag, setFlag] = useState(false);
    const [btn, setBtn] = useState([]);
    const [nameButtons, setNameButtons] = useState([]);
    const [loading, setLoading] = useState(false);
    const [params, setParams] = useState([]);
    const [url, setUrl] = useState("");
    const [hideMenu, setHideMenu] = useState(false);

    const urlImg = "https://assets.imgix.net/unsplash/";

    //consume photos API
    useEffect(() => {
        setLoading(true);
        try {
            axios("https://storage.googleapis.com/nanlabs-engineering-technical-interviews/imgix-samples-list.json")
                .then((res) => {
                    setData(res.data);
                })
        } catch (error) {
            console.log(error);
        }
        setLoading(false);
    }, [])

    //Pass data from the params to the state of btn so that the names appear in the buttons
    useEffect(() => {
        setNameButtons(Params);
    }, [])


    //Pass the parameters of the button that you click
    const addControl = (v) => {
        let parametros = [...Params]
        parametros.forEach((param) => {
            if (v === param.alias) {
                setBtn(param)
            }
        })
        setFlag(true);
    }

    // filter the data of the params to be able to take a single value per parameter
    // and build the url with all the effects we want in the photos
    const addParamUrl = (alias, value) => {
        let aux = [...params];
        let isParam = false;

        aux.forEach((a) => {
            if (a.alias === alias) {
                isParam = true;
                a.value = value;
                setParams(aux);
            }
        })
        !isParam &&
            setParams([...aux, { alias, value }]);
        convertUrl();

    }

    //Convert the array of objects into a string to use as the url of the image
    const convertUrl = () => {
        let stringP = "";
        for (let i = 0; i < params.length; i++) {
            stringP = stringP + `${params[i].alias}=${params[i].value}&`

        }
        setUrl(stringP);

    }

    //reset url with change img
    const resetUrl = () => {
        setUrl("");
        setParams([]);
        setFlag(false);
    }

    //delete the last parameter added
    const removeItem = () => {
        params.pop();
        convertUrl();
        setFlag(false);
    }

    return (
        <DataContext.Provider
            value={{
                data,
                setData,
                addControl,
                nameButtons,
                flag,
                btn,
                loading,
                params,
                setParams,
                addParamUrl,
                url,
                removeItem,
                hideMenu,
                setHideMenu,
                resetUrl,
                urlImg
            }}>
            {props.children}
        </DataContext.Provider>
    )
}

export default DataContextProvider;
