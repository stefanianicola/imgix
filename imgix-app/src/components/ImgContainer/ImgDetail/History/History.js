import React, { useContext } from 'react';
import './History.scss';
import { DataContext } from '../../../../context/DataContext';
import { Button, Row, Col } from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';

const History = () => {
    const { removeItem } = useContext(DataContext)

    return (
        <div>
            <Row className="row-prev">
                <Col xs={3}>
                <Button  onClick={removeItem}>
                    <i className="fa fa-angle-left" aria-hidden="true"></i>
                Prev
                </Button>
                </Col>
            </Row>
        </div>
    )
}
export default History;