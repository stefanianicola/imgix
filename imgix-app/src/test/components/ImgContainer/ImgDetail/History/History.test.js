import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import History from '../../../../../components/ImgContainer/ImgDetail/History/History'
import { DataContext } from '../../../../../context/DataContext';

describe('History component', () => {
    const removeItem = jest.fn()
    const wrapper = mount(
        <Router>
            <DataContext.Provider value={{ removeItem }} >
                <History />
            </DataContext.Provider>
        </Router>
    );


    it('should be create the component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should eject removeItem function', () => {
        const btn = wrapper.find('button');
        btn.simulate('click');

        expect(removeItem).toHaveBeenCalled();
    });

})