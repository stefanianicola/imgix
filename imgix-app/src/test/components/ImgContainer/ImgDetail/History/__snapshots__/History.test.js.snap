// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History component should be create the component 1`] = `
<div>
  <div
    className="row-prev row"
  >
    <div
      className="col-3"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[MockFunction]}
        type="button"
      >
        <i
          aria-hidden="true"
          className="fa fa-angle-left"
        />
        Prev
      </button>
    </div>
  </div>
</div>
`;
